packages/components/src/components/page-error/__tests__/page-error.spec.tsx

```diff
@@ -4,44 +4,148 @@ import { screen, render } from '@testing-library/react';
-        const setError = jest.fn();
-        renderWithRouter(<PageError {...props} should_redirect={true} setError={setError} />);
+
+    it('Should render the redirect button correctly', () => {
+        renderWithRouter(<PageError {...{ ...pageErrorDefaultProps, redirect_urls: ['/'] }} should_redirect={true} />);
+
+        const button = screen.getByText('Go back');
+        expect(button).toBeInTheDocument();
+    });
+
+    it('Should call setError() when redirect button gets clicked', () => {
+        const { setError } = pageErrorDefaultProps;
+        renderWithRouter(<PageError {...pageErrorDefaultProps} should_redirect={true} />);
+
         const link = screen.getByRole('link');
+
         userEvent.click(link);
         expect(setError).toHaveBeenCalledTimes(1);
     });
+
+    it('Should call setError() when redirect button gets clicked and should_redirect did not pass', () => {
+        const { setError } = pageErrorDefaultProps;
+        renderWithRouter(<PageError {...pageErrorDefaultProps} should_redirect={undefined} />);
+
+        const link = screen.getByRole('link');
+
+        userEvent.click(link);
+        expect(setError).toHaveBeenCalled();
+    });
+
+    it('Should call setError when button is clicked and should_clear_error_on_click is true', () => {
+        renderWithRouter(<PageError {...pageErrorDefaultProps} />);
+
+        const button = screen.getByText('Go back');
+        userEvent.click(button);
+
+        expect(mockSetError).toHaveBeenCalledWith(false, null);
+    });
+
+    it('Should not show the button when redirect equals to false and redirect_labels has nothing to display', () => {
+        const { buttonOnClick } = pageErrorDefaultProps;
+        renderWithRouter(
+            <PageError
+                {...{ ...pageErrorDefaultProps, redirect_labels: [''] }}
+                should_clear_error_on_click={false}
+                buttonOnClick={buttonOnClick}
+                should_redirect={false}
+            />
+        );
+
+        const page_error = screen.getByTestId('dc-page-error__btn-wrapper');
+
+        expect(page_error).not.toHaveTextContent();
+    });
+
+    it('Should show on mobile version text size of the error equals to "xs" and not show the button when redirect equals to false and redirect_labels has one element in the array', () => {
+        const { buttonOnClick } = pageErrorDefaultProps;
+
+        (isMobile as jest.Mock).mockReturnValueOnce(true);
+
+        renderWithRouter(
+            <PageError
+                {...{ ...pageErrorDefaultProps, redirect_labels: ['Some redirect label'] }}
+                should_clear_error_on_click={false}
+                buttonOnClick={buttonOnClick}
+                should_redirect={false}
+            />
+        );
+
+        const spanElement = screen.getByText('Some redirect label');
+
+        expect(spanElement).toBeInTheDocument();
+        expect(spanElement).toHaveStyle(`--text-size: var(--text-size-xs)`);
```

Comment:
name convention - snake case ;) but it's better to name it `redirect_label` instead of span_element

Reply:
fixed✅